ddl/column.go

```diff
@@ -690,6 +690,12 @@ func needChangeColumnData(oldCol, newCol *model.ColumnInfo) bool {
 	needTruncationOrToggleSign := func() bool {
 		return newCol.Flen > 0 && newCol.Flen < oldCol.Flen || toUnsigned != originUnsigned
 	}
+	// Ignore the potential max display length represented by integer's flen.
+	oldColFlen, _ := mysql.GetDefaultFieldLengthAndDecimal(oldCol.Tp)
+	newColFlen, _ := mysql.GetDefaultFieldLengthAndDecimal(newCol.Tp)
+	needTruncationOrToggleSignForInteger := func() bool {
+		return newColFlen > 0 && newColFlen < oldColFlen || toUnsigned != originUnsigned
+	}
```

Comment:
Why not use a variable instead?

Reply:
Cause for most of the cases, we don't need to trigger the computation of it, like a kind of lazy call if necessary?